secboot/secboot_sb_test.go

```diff
@@ -605,15 +605,61 @@ func (s *secbootSuite) TestEFIImageFromBootFile(c *C) {
 	}
 }
 
+func (s *secbootSuite) TestProvisionTPM(c *C) {
+	mockErr := errors.New("some error")
+
+	for idx, tc := range []struct {
+		tpmErr            error
+		tpmEnabled        bool
+		provisioningErr   error
+		provisioningCalls int
+		expectedErr       string
+	}{
+		{tpmErr: mockErr, expectedErr: "cannot connect to TPM: some error"},
+		{tpmEnabled: false, expectedErr: "TPM device is not enabled"},
+		{tpmEnabled: true, provisioningErr: mockErr, provisioningCalls: 1, expectedErr: "cannot provision TPM: some error"},
+		{tpmEnabled: true, provisioningCalls: 1, expectedErr: ""},
+	} {
+		c.Logf("tc: %v", idx)
+		d := c.MkDir()
+		tpm, restore := mockSbTPMConnection(c, tc.tpmErr)
+		defer restore()
+
+		// mock TPM enabled check
+		restore = secboot.MockIsTPMEnabled(func(t *sb_tpm2.Connection) bool {
+			return tc.tpmEnabled
```

Comment:
a counter might be good here too (unless invoking this method is optional, in which case don't bother -- it would just make the test harder to read)

Reply:
the mocked secboot provision function will not be called unless TPM is enabled which is already verified by the expected number of calls